### DISCUSSION TITLE
Manual installation of Open webui failed

### DISCUSSION THREAD

**Discussion by Chun-QiuCC** [open-webui/open-webui#1529] [General]
# Bug Report

## Description

**Bug Summary:**

pip reports an error during manual installation according to https://docs.openwebui.com/getting-started/

**Steps to Reproduce:**
Use pip 24.0 and python3.11 to execute the following instructions
```
git clone https://github.com/open-webui/open-webui.git
cd open-webui/

# Copying required .env file
cp -RPp .env.example .env

# Building Frontend Using Node
npm i
npm run build

#Serving Frontend with the Backend
cd ./backend
pip install -r requirements.txt -U
bash start.sh
```
This step pip install -r requirements.txt -U reports an error

**Expected Behavior:**

No errors reported, normal installation

**Actual Behavior:**

pip报错，信息如下：
```
root@ollama:~/open-webui/backend# pip install -r requirements.txt -U -i https://mirrors.aliyun.com/pypi/simple/
Looking in indexes: https://mirrors.aliyun.com/pypi/simple/
Collecting fastapi (from -r requirements.txt (line 1))
  Downloading https://mirrors.aliyun.com/pypi/packages/c0/c1/2dc286475c8e2e455e431a1cf1cf29662c9f9290434161088ba039d77481/fastapi-0.110.1-py3-none-any.whl (91 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 91.9/91.9 kB 982.2 kB/s eta 0:00:00
Collecting pydantic (from -r requirements.txt (line 3))
  Downloading https://mirrors.aliyun.com/pypi/packages/16/ca/330c4f3bd983bb24ac12c7fd1e08c26c8aed70bc64498cf38c770321067f/pydantic-2.7.0-py3-none-any.whl (407 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 407.9/407.9 kB 955.2 kB/s eta 0:00:00
Collecting python-multipart (from -r requirements.txt (line 4))
  Downloading https://mirrors.aliyun.com/pypi/packages/3d/47/444768600d9e0ebc82f8e347775d24aef8f6348cf00e9fa0e81910814e6d/python_multipart-0.0.9-py3-none-any.whl (22 kB)
Collecting flask (from -r requirements.txt (line 6))
  Downloading https://mirrors.aliyun.com/pypi/packages/61/80/ffe1da13ad9300f87c93af113edd0638c75138c42a0994becfacac078c06/flask-3.0.3-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.7/101.7 kB 723.7 kB/s eta 0:00:00
Collecting flask_cors (from -r requirements.txt (line 7))
  Downloading https://mirrors.aliyun.com/pypi/packages/10/69/1e6cfb87117568a9de088c32d6258219e9d1ff7c131abf74249ef2031279/Flask_Cors-4.0.0-py2.py3-none-any.whl (14 kB)
Collecting python-socketio (from -r requirements.txt (line 9))
  Downloading https://mirrors.aliyun.com/pypi/packages/af/bf/be12875b17709b591d8505811e513a88b31316e4ce0e801da351b4765ea5/python_socketio-5.11.2-py3-none-any.whl (75 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 75.7/75.7 kB 896.2 kB/s eta 0:00:00
Collecting python-jose (from -r requirements.txt (line 10))
  Downloading https://mirrors.aliyun.com/pypi/packages/bd/2d/e94b2f7bab6773c70efc70a61d66e312e1febccd9e0db6b9e0adf58cbad1/python_jose-3.3.0-py2.py3-none-any.whl (33 kB)
Collecting uuid (from -r requirements.txt (line 12))
  Downloading https://mirrors.aliyun.com/pypi/packages/ce/63/f42f5aa951ebf2c8dac81f77a8edcc1c218640a2a35a03b9ff2d4aa64c3d/uuid-1.30.tar.gz (5.8 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

## Environment

- **Operating System:** ubuntu-20.04.6-live-server-amd64.iso
- **Browser (if applicable):** none

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- Removed Docker and browser options since it was a manual installation and the installation failed

## Logs and Screenshots

**Browser Console Logs:**
none

**Docker Container Logs:**
none (manual installation)

**Screenshots (if applicable):**
![error-img](https://ollama.tyloo.pro/imgs/open-webui-pip-error.png)

```
root@ollama:~/open-webui/backend# pip install -r requirements.txt -U -i https://mirrors.aliyun.com/pypi/simple/
Looking in indexes: https://mirrors.aliyun.com/pypi/simple/
Collecting fastapi (from -r requirements.txt (line 1))
  Downloading https://mirrors.aliyun.com/pypi/packages/c0/c1/2dc286475c8e2e455e431a1cf1cf29662c9f9290434161088ba039d77481/fastapi-0.110.1-py3-none-any.whl (91 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 91.9/91.9 kB 982.2 kB/s eta 0:00:00
Collecting pydantic (from -r requirements.txt (line 3))
  Downloading https://mirrors.aliyun.com/pypi/packages/16/ca/330c4f3bd983bb24ac12c7fd1e08c26c8aed70bc64498cf38c770321067f/pydantic-2.7.0-py3-none-any.whl (407 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 407.9/407.9 kB 955.2 kB/s eta 0:00:00
Collecting python-multipart (from -r requirements.txt (line 4))
  Downloading https://mirrors.aliyun.com/pypi/packages/3d/47/444768600d9e0ebc82f8e347775d24aef8f6348cf00e9fa0e81910814e6d/python_multipart-0.0.9-py3-none-any.whl (22 kB)
Collecting flask (from -r requirements.txt (line 6))
  Downloading https://mirrors.aliyun.com/pypi/packages/61/80/ffe1da13ad9300f87c93af113edd0638c75138c42a0994becfacac078c06/flask-3.0.3-py3-none-any.whl (101 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.7/101.7 kB 723.7 kB/s eta 0:00:00
Collecting flask_cors (from -r requirements.txt (line 7))
  Downloading https://mirrors.aliyun.com/pypi/packages/10/69/1e6cfb87117568a9de088c32d6258219e9d1ff7c131abf74249ef2031279/Flask_Cors-4.0.0-py2.py3-none-any.whl (14 kB)
Collecting python-socketio (from -r requirements.txt (line 9))
  Downloading https://mirrors.aliyun.com/pypi/packages/af/bf/be12875b17709b591d8505811e513a88b31316e4ce0e801da351b4765ea5/python_socketio-5.11.2-py3-none-any.whl (75 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 75.7/75.7 kB 896.2 kB/s eta 0:00:00
Collecting python-jose (from -r requirements.txt (line 10))
  Downloading https://mirrors.aliyun.com/pypi/packages/bd/2d/e94b2f7bab6773c70efc70a61d66e312e1febccd9e0db6b9e0adf58cbad1/python_jose-3.3.0-py2.py3-none-any.whl (33 kB)
Collecting uuid (from -r requirements.txt (line 12))
  Downloading https://mirrors.aliyun.com/pypi/packages/ce/63/f42f5aa951ebf2c8dac81f77a8edcc1c218640a2a35a03b9ff2d4aa64c3d/uuid-1.30.tar.gz (5.8 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

## Installation Method

manual installation

## Additional Information

none

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by vibora** [comment]
I'm by far no expert, but did you try
```python
pip install open-webui
``` 

And afterwards 
```python
open-webui server
``` 
I also got problems with the manual installation method but the pip installation helped to get open-webui up and running

**Reply by ShikoThePro** [reply]
Yes I tried this method but I got this error

C:\Users\SHIKO>open-webui serve
Loading WEBUI_SECRET_KEY from file, not provided as an environment variable.
Loading WEBUI_SECRET_KEY from C:\Users\SHIKO\.webui_secret_key
0
True
┌─────────────────────────────── Traceback (most recent call last) ────────────────────────────────┐
│ C:\Users\SHIKO\AppData\Local\Programs\Python\Python311\Lib\site-packages\open_webui\__init__.py: │
│ 45 in serve                                                                                      │
│                                                                                                  │
│   42 │   │   )                                                                                   │
│   43 │   import main  # we need set environment variables before importing main                  │
│   44 │                                                                                           │
│ > 45 │   uvicorn.run(main.app, host=host, port=port, forwarded_allow_ips="*")                    │
│   46                                                                                             │
│   47                                                                                             │
│   48 @app.command()                                                                              │
│                                                                                                  │
│ ┌─────────────────────────────────────────── locals ───────────────────────────────────────────┐ │
│ │ host = '0.0.0.0'                                                                             │ │
│ │ main = <module 'main' from                                                                   │ │
│ │        'C:\\Users\\SHIKO\\AppData\\Local\\Programs\\Python\\Python311\\main.py'>             │ │
│ │ port = 8080                                                                                  │ │
│ └──────────────────────────────────────────────────────────────────────────────────────────────┘ │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
AttributeError: module 'main' has no attribute 'app'

**Comment by JuanjoJG-WATA** [comment]
That happened to me. Installing node.js like this (OPTION 3) did the trick: 

[How To Install Node.js on Ubuntu 20.04](https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-20-04)

Also, maybe try it with pip3 instead of only pip:
`pip3 install -r requirements.txt -U`

**Reply by Chun-QiuCC** [reply]
Thanks, I have successfully built the webui using docker-compose, I will try this method later

**Reply by JuanjoJG-WATA** [reply]
Hope it works, cheers!

**Comment by rogical** [comment]
Same issue

Collecting psycopg2-binary==2.9.9 (from -r requirements.txt (line 17))
  Using cached psycopg2-binary-2.9.9.tar.gz (384 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.

**Comment by Areshkew** [comment]
I am getting:


`OperationalError: (sqlite3.OperationalError) no such table: config
[SQL: SELECT config.id AS config_id, config.data AS config_data, config.version AS config_version,
config.created_at AS config_created_at, config.updated_at AS config_updated_at
FROM config ORDER BY config.id DESC
 LIMIT ? OFFSET ?]
[parameters: (1, 0)]`

**Reply by rastafarra** [reply]
i have same error, how i can fix it?
